app/grandchallenge/archives/views.py

```diff
@@ -558,6 +564,128 @@ def get_serializer_class(self):
             return ArchiveItemSerializer
 
 
+class ArchiveItemJobListView(
+    LoginRequiredMixin,
+    ObjectPermissionRequiredMixin,
+    PaginatedTableListView,
+):
+    model = Job
+    template_name = "archives/archive_item_job_list.html"
+    row_template = "archives/archive_item_job_list_row.html"
+
+    search_fields = [
+        "pk",
+        "creator__username",
+        "comment",
+        # Inputs
+        "inputs__image__name",
+        "inputs__image__files__file",
+        # Outputs
+        "outputs__interface__title",
+        "outputs__interface__description",
+        # Algorithm
+        "algorithm_image__algorithm__title",
+        "algorithm_image__algorithm__description",
+        "algorithm_image__comment",
+    ]
+
+    permission_required = (
+        f"{Archive._meta.app_label}.view_{ArchiveItem._meta.model_name}"
+    )
+    raise_exception = True
+
+    login_url = reverse_lazy("account_login")
+
+    columns = [
+        *JobsList.columns[:2],
+        Column(
+            title="Algorithm", sort_field="algorithm_image__algorithm__title"
+        ),
+        *JobsList.columns[3:],
+    ]
+    default_sort_column = JobsList.default_sort_column
+
+    @cached_property
+    def archive_item(self):
+        return get_object_or_404(ArchiveItem, pk=self.kwargs["pk"])
+
+    def get_permission_object(self):
+        return self.archive_item
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        input_civs = self.archive_item.values.all()
+        num_civs = len(input_civs)
+
+        queryset = (
+            queryset.filter(
+                pk__in=Subquery(
+                    Job.objects.filter(
+                        inputs__archive_items=self.archive_item
+                    ).values("pk"),
+                )
+            )
```

Comment:
For my understanding: why do you need a subquery here? Can't you just do something like `queryset.filter(inputs__archive_items=self.archive_item)`? And why is it faster than doing `inputs__in=[input_civs]`?

Reply:
I do get your confusion! The need for this rolled out of the query and filter tests I wrote that this is needed.  I think the following is happening:

`queryset.filter(inputs__archive_items=self.archive_item)` would result in a join that contains rows of Jobs and CIVs. Each Job is represented as many times as it has CIVs **that match the archvie_item**. However, it would not have a representation of a CIV that IS an input but not part of the archive item. Which would then result in incorrect detection by the `WHERE`/`annotate() + filter()` and might include subsets.

Rather what we need is all the jobs that _contain_  an output that is also in the archive item. Hence the subquery and selection on `pk`. 

There might be a different approach, but this passed the tests and seemed performant via an Index Scan in the query planner + practical local tests.